scripts/utils.sh

```diff
@@ -0,0 +1,384 @@
+#######################################
+
+#######################################
+# Add new commented entry in server hosts file.
+# Arguments:
+#   1) IP address to add
+#   2) Corresponding Fully Qualified Domain Name to add
+#   3) Corresponding entry comment to add
+# Outputs:
+#   Writes log to STDOUT.
+#######################################
+function utils::add_hosts_file_entry() {
+  local ip="$1"
+  local fqdn="$2"
+  local comment="$3"
+
+  readonly HOSTS_FILE_PATH="/etc/hosts"
+
+  utils::echo_action "Adding entry for ${fqdn} in ${HOSTS_FILE_PATH}..."
+  if ! grep -q "${fqdn}" "${HOSTS_FILE_PATH}"; then
+    printf "# ${comment}\n${ip} ${fqdn}\n" >> "${HOSTS_FILE_PATH}"
+    utils::echo_info "Done."
+  else
+    utils::echo_info "Skipped: ${HOSTS_FILE_PATH} already contains entry for ${fqdn}."
+  fi
+}
+
+#######################################
+# Echo a text string using the ACTION format.
+# Globals:
+#   DATE_FORMAT
+# Arguments:
+#   Message to echo
+# Outputs:
+#   Writes message to echo to STDOUT.
+#######################################
+function utils::echo_action() {
+  echo ""
+  echo "$(date +"$DATE_FORMAT") | ACTION - $1"
+}
+
+#######################################
+# Echo a text string using the ERROR format.
+# Globals:
+#   DATE_FORMAT
+# Arguments:
+#   Message to echo
+# Outputs:
+#   Writes message to echo to STDERR.
+#######################################
+function utils::echo_error() {
+  echo "$(date +"$DATE_FORMAT") | ERROR  - $1" >&2
+}
+
+#######################################
+# Echo a text string using the INFO format.
+# Globals:
+#   DATE_FORMAT
+# Arguments:
+#   Message to echo
+# Outputs:
+#   Writes message to echo to STDOUT.
+#######################################
+function utils::echo_info() {
+  echo "$(date +"$DATE_FORMAT") | INFO   - $1"
+}
+
+#######################################
+# Echo a text string using the TITLE format.
+# Arguments:
+#   Message to echo
+# Outputs:
+#   Writes message to echo to STDOUT.
+#######################################
+function utils::echo_title() {
+  echo ""
+  echo "###############################################################################"
+  echo "$1"
+  echo "###############################################################################"
+}
+
+#######################################
+# Harden Apache2 Webserver settings.
+# Arguments:
+#   Apache2 security configuration file to update, a file path.
+# Outputs:
+#   Writes normal log messages to STDOUT.
+#   Writes error messages to STDERR.
+#######################################
+function utils::harden_apache2() {
+  local config_file_path="$1"
+
+  utils::update_apache2_config_file "ServerTokens" "Prod" "${config_file_path}"
+  utils::update_apache2_config_file "ServerSignature" "Off" "${config_file_path}"
+}
+
+#######################################
+# Mount a data disk identified by its size to a mount point.
+# Since that is no way to predict the device name of a data disk, we use the
```

Comment:
That sentence might need some improvement

Reply:
Corrigé.